src/Microsoft.Azure.WebJobs.Logging/Entities/RecentPerFuncEntity.cs

```diff
@@ -108,20 +114,23 @@ public string FunctionInstanceId
         public string FunctionName { get; set; }
 
         public string DisplayName { get; set; }
-
-        public string RawStatus { get; set;  }
-
-        FunctionInstanceStatus IFunctionInstanceBaseEntry.Status
+        
+        public bool HasError
         {
             get
             {
-                FunctionInstanceStatus e;
-                if (!Enum.TryParse<FunctionInstanceStatus>(this.RawStatus, out e))
-                {
-                    return FunctionInstanceStatus.Unknown;
-                }
-                return e;
+                // Use RawStatus for legacy check. 
+                return _hasError || (this.RawStatus == "CompletedFailed");
+            }
+            set
+            {
+                _hasError = value;
             }
         }
+
+        /// <summary>
+        ///  Legacy for errors. We only read from this to check if it's "CompletedFailed" so we pick up legacy errors. 
```

Comment:
I'll clarify the comment. 
        ///  Original ("legacy") log writers would write out the function status. 
        ///  Now, we write separate bits (start time, end time, HasError) and infer the status from that. 
        ///  However, for the legacy case, it didn't have HasError and so we need to detect an error if the status field was set to "CompletedFailed".
        ///  New cases don't even write this status field anymore. 

---
In reply to: [90489837](https://github.com/Azure/azure-webjobs-sdk/pull/928#discussion_r90489837) [](ancestors = 90489837)